### PR TITLE
Move db_access to the new ugbio_utils structure

### PR DESCRIPTION
**Pull request by ilyasoifer**
Moved db_access to ugbio-utils. Related to Ultimagen/VariantCalling#206

### CODE REVIEW

**Review by TammushB — APPROVED** [review]
LGTM

**Comment by ilyasoifer** [comment]
@avigailmo and @inbalzelig-ugi - There is a crash of CI that I don't know how to fix. Can you advise please?

**Comment by avigailmo** [comment]
> @avigailmo and @inbalzelig-ugi - There is a crash of CI that I don't know how to fix. Can you advise please?

the quash wasn't in CI - but in other action that I fixed today. I rebased to latest master.

**Review by avigailmo — COMMENTED** [review]
looks good
Your new workspace is not included in the ci and publish builds. (We have an opened task to make the workspaces list dynamic- but now it's static [here](https://github.com/Ultimagen/ugbio-utils/blob/main/.github/workflows/ci.yml#L11) and [here](https://github.com/Ultimagen/ugbio-utils/blob/main/.github/workflows/publish.yml#L101)) Should add them (and add a Dockerfile)

Also, need to add a branch in variant calling that deletes ugvc db_access and update consumers to use ug_db instead

File: src/db/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[project]
+name = "ugbio_db"
+version = "1.1.0.dev0"
+requires-python = ">= 3.11"
+dependencies = [
+    "ugbio_core",
```
Comment (inbalzelig-ugi) [inline src/db/pyproject.toml:6]: You don't need the core at all. I think once you remove it you will need to add pandas. But that is it.

**Comment by ilyasoifer** [comment]
> @avigailmo @inbalzelig-ugi why is db_access in a different module? I think it should be in core (common utils). also, when moving content to core I think we should keep original folders naming so we will keep some kind of ordering.

@avigailmo, @TammushB - happy to go either way, please let me know what fits better

File: src/db/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,53 @@
+FROM python:3.10-bullseye as build
+
+ARG OUT_DIR=/tmp/ugbio
+RUN mkdir -p $OUT_DIR
+
+COPY ./src/ ./src/
+
+#build ugbio_core
+RUN python -m pip install build
+RUN python -m build --outdir $OUT_DIR --wheel ./src/core
```
Comment (inbalzelig-ugi) [inline src/db/Dockerfile:10]: You don't need core in your docker. Instead you are missing your package code. See other dockerfiles in the repo as a reference and remove all packages you don't need.

File: src/db/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    "pymongo>=4.8.0",
+]
+description = "Ultima Genomics Internal DB Access utils"
+authors = [
+    { name = "Ilya Soifer", email = "ilya.soifer@ultimagen.com" },
+]
+readme = "README.db.md"
```
Comment (inbalzelig-ugi) [inline src/db/pyproject.toml:13]: You can add scripts to the pyprojct.toml which can be used as entry points in the docker. This is very nice and clean, consider adding it here as well

**Review by inbalzelig-ugi — CHANGES_REQUESTED** [review]
That is a good start! The dockerfile and the env need some more work. 
You are missing an action to build the docker.

File: src/db/Dockerfile (hunk 2 above)
Comment (ilyasoifer) [reply src/db/Dockerfile:11]: Fixed

File: src/db/pyproject.toml (hunk 3 above)
Comment (ilyasoifer) [reply src/db/pyproject.toml:13]: True. But I don't have scripts in this module.

**Comment by avigailmo** [comment]
> > @avigailmo @inbalzelig-ugi why is db_access in a different module? I think it should be in core (common utils). also, when moving content to core I think we should keep original folders naming so we will keep some kind of ordering.
> 
> @avigailmo, @TammushB - happy to go either way, please let me know what fits better

As we discussed yesterday db_access is never used by any wdl but used for research. If we add it to core - this script + its dv dependencies will be inserted into all of our ugbio dockers.

**Comment by avigailmo** [comment]
> That is a good start! The dockerfile and the env need some more work. You are missing an action to build the docker.

I don't think we need an action to build the docker
Currently it used only by tests

**Comment by ilyasoifer** [comment]
@avigailmo, @inbalzelig-ugi  The CI mostly works now. Now I want to disable tests in test_db_access that crash because the runner does not have access to Papyrus. Can you suggest how to do this?

File: src/db/pyproject.toml (hunk 1 above)
Comment (ilyasoifer) [reply src/db/pyproject.toml:6]: Fixed

**Comment by ilyasoifer** [comment]
> looks good Your new workspace is not included in the ci and publish builds. (We have an opened task to make the workspaces list dynamic- but now it's static [here](https://github.com/Ultimagen/ugbio-utils/blob/main/.github/workflows/ci.yml#L11) and [here](https://github.com/Ultimagen/ugbio-utils/blob/main/.github/workflows/publish.yml#L101)) Should add them (and add a Dockerfile)
> 
> Also, need to add a branch in variant calling that deletes ugvc db_access and update consumers to use ug_db instead

Mostly done. https://github.com/Ultimagen/VariantCalling/pull/206 is the branch in VariantCalling I believe

**Comment by ilyasoifer** [comment]
@avigailmo, @inbalzelig-ugi  - I would be happy for your advice please
Thanks!

**Comment by ilyasoifer** [comment]
closing this for now, will be submitted at a later stage

**Pull request closed without merging**